test/integration/policy_certificate_test.go

```diff
@@ -25,122 +28,147 @@ import (
+			)
+			Expect(err).To(BeNil())
+
+			By("Checking that " + policyCertificateName + " exists on the Hub cluster")
+			rootPolicy := utils.GetWithTimeout(
+				clientHubDynamic, common.GvrPolicy, policyCertificateName, userNamespace, true, defaultTimeoutSeconds,
+			)
+			Expect(rootPolicy).NotTo(BeNil())
+		})
+
+		It("stable/"+policyCertificateName+" should be created on managed cluster", func() {
+			By("Checking the policy on the managed cluster in the namespace " + clusterNamespace)
+			managedPolicy := utils.GetWithTimeout(
+				clientManagedDynamic,
+				common.GvrPolicy,
+				userNamespace+"."+policyCertificateName,
+				clusterNamespace,
+				true,
+				defaultTimeoutSeconds,
+			)
+			Expect(managedPolicy).NotTo(BeNil())
+		})
+
+		It("stable/"+policyCertificateName+" should be Compliant", func() {
+			By("Checking if the status of the root policy is Compliant")
+			Eventually(
+				common.GetComplianceState(clientHubDynamic, userNamespace, policyCertificateName, clusterNamespace),
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(Equal(policiesv1.Compliant))
+		})
+
+		It("Make the policy NonCompliant", func() {
+			By("Creating a secret with an expired certificate")
+			key, err := rsa.GenerateKey(rand.Reader, 2048)
+			Expect(err).To(BeNil())
+
+			template := x509.Certificate{
+				SerialNumber: big.NewInt(1),
+				Subject: pkix.Name{
+					Organization: []string{"Red Hat"},
+				},
+				NotBefore:             time.Now().Add(time.Hour * -25),
+				NotAfter:              time.Now().Add(time.Hour * -1),
+				DNSNames:              []string{"www.example.com"},
+				KeyUsage:              x509.KeyUsageDataEncipherment | x509.KeyUsageDigitalSignature,
+				ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+				BasicConstraintsValid: true,
+			}
+			derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &key.PublicKey, key)
+			Expect(err).To(BeNil())
+
+			pemBytes := &bytes.Buffer{}
+			err = pem.Encode(pemBytes, &pem.Block{Type: "CERTIFICATE", Bytes: derBytes})
+			Expect(err).To(BeNil())
+
+			secret := corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: expiredCertSecretName},
+				Data:       map[string][]byte{"tls.crt": pemBytes.Bytes()},
+			}
+			_, err = clientManaged.CoreV1().Secrets(policyCertificateNSName).Create(
+				context.TODO(), &secret, metav1.CreateOptions{},
+			)
+			Expect(err).To(BeNil())
+		})
+
+		It("stable/"+policyCertificateName+" should be NonCompliant", func() {
+			By("Checking if the status of the root policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(clientHubDynamic, userNamespace, policyCertificateName, clusterNamespace),
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+		})
+
+		AfterAll(func() {
+			_, err := utils.KubectlWithOutput(
+				"delete", "-f", policyCertificateURL, "-n", userNamespace, "--kubeconfig="+kubeconfigHub,
+			)
+			if !k8serrors.IsNotFound(err) {
```

Comment:
I don't think `k8serrors.IsNotFound` will work since the error is not an API call but from the CLI.

Instead, please use the `--ignore-not-found` flag in the `kubectl delete` command.